docs/tutorials/broadcast-join.rst

```diff
@@ -0,0 +1,298 @@
+GeoMesa Spark: Aggregating Data
+===============================
+
+This tutorial will show you how to:
+
+1. Use GeoMesa with `Apache Spark <http://spark.apache.org/>`__ in Scala.
+2. Create and use DataFrames with our geospatial User Defined Functions.
+3. Calculate aggregate statistics using a covering set of polygons.
+4. Create a new simple feature type to represent this aggregation.
+5. Visualize the result as a choropleth.
+
+Background
+----------
+
+`GDELT <http://gdeltproject.org/>`__ provides a comprehensive time- and location-indexed archive of events reported
+in broadcast, print, and web news media worldwide from 1979 to today.
+
+`FIPS Codes <https://www.census.gov/geo/maps-data/data/cbf/cbf_counties.html>`__ are
+Federal Information Processing Standard Publication codes that uniquely identify counties in the United States.
+
+If we make the assumption that a county with more GDELT events is somehow more politically relevant, we may be interested
+in seeing which counties are more dense with GDELT events.
+
+Since GDELT is a data set with point geometries, we do not immediately know which county it belongs to. To resolve this,
+we will use a second data set, a FIPS Codes shapefile that denotes the boundary and FIPS code of a county, and join GDELT
+points with the county that contains them.
+
+In this case, the number of counties is particularly small, approximately 3000 records, we can make our query much
+more efficient by "broadcasting" the counties. "Broadcast" here meaning a
+`Spark Broadcast <https://spark.apache.org/docs/2.2.0/api/java/org/apache/spark/sql/functions.html#broadcast-org.apache.spark.sql.Dataset->`__.
+In a traditional Spark SQL join, data will be shuffled around the executors based on the partitioners of the RDDs,
+and since in our case the join key is a geometric field, there is no Spark partitioner that can map data effectively.
+The resulting movement of data across nodes is expensive, so we can attain a performance boost by sending (broadcasting)
+our entire small data set to each of the nodes once. This ensures that the executors have all the data needed to compute
+the join, and no additional shuffling is needed.
+
+Spark provides a means of executing this "broadcast join", though it should only be used when the data being broadcast
+is small enough to fit in the memory of the executors.
+
+Prerequisites
+-------------
+
+For this tutorial, we will assume that your have already ingested the two data sets into the data store of your choosing.
+Following this tutorial without having created the necessary tables will lead to errors.
+
+If you have not ingested any data, you can follow one of the ingest tutorials :doc:`/tutorials/geomesa-examples-gdelt`.
+The converter for the GDELT data set, :doc:`/user/convert/premade/gdelt`, is provided with GeoMesa, and the FIPS data can
+be ingested without a converter as a shapefile.
```

Comment:
I'd change the order of these two paragraphs, and rewrite to have this kind of flow:

"One can ingest the data using converter, etc, etc, and FIPS data is available in shapefile."

"Once you have the data ingested in GeoMesa, you can continue with the rest of the tutorial..."